### ISSUE TITLE
how to covert a png image to geometry data?

### ISSUE THREAD

**Issue by wkingnet** [Tugcga/Path-Finder#15]
I have a black-white png image, and I can convert image data to np.array data.

So now how to convert np.array to geometry data like this?
```
baker.add_geometry([(-1.0, 0.0, -1.0), (-1.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 0.0, -1.0), (-1.0, 1.0, -1.0), (-1.0, 1.0, 1.0), (1.0, 1.0, 1.0), (1.0, 1.0, -1.0)],[[0, 3, 2, 1], [2, 6, 5, 1], [4, 5, 6, 7], [0, 4, 7, 3], [2, 3, 7, 6], [0, 1, 5, 4]])
```

Thanks a lot.

**Comment by wkingnet** [comment]
> It's not clear what data your png image contains. And how you want to convert it to geometry. Maybe there is an example? What you have and what you want to get.

![image](https://github.com/Tugcga/Path-Finder/assets/57652989/40fa9dce-9020-457b-bd15-7f9bfbac818a)

I have a game world map, and this world map is different from the real world map. 

I want to plan some sea routes.

I have tried the JPS algorithm, but after testing I found that the JPS algorithm does not suit for my case. JPS algorithm is too slow for me, it takes 60 seconds to find a path. Even using JPS+ it takes 5 seconds. And I need to give the result within 1 second.

**Comment by Tugcga** [comment]
You can try to convert your map into mesh as follows. Ignore black pixels, but for each white pixel at ```x``` column and ```y``` row create a square polygon and add it to the baker
```
baker.add_geometry([(x, 0.0, y), (x, 0.0, y + 1.0), (x + 1.0, 0.0, y + 1.0), (x + 1.0, 0.0, y)], [[0, 1, 2, 3]])
```

Then chose suitable bake parameters (mostly ```cell_size``` and ```agent_radius```) and bake the navmesh.

**Comment by wkingnet** [comment]
I don't think this method will work. I added all the white points and the code ran for 2 hours and used up all the memory. I think it is impossible for the code to return results.

![image](https://github.com/Tugcga/Path-Finder/assets/57652989/ef3f1ddd-d75f-4f51-84eb-7277257b317e)


```
import numpy as np
from tqdm import tqdm
from pathfinder import navmesh_baker as nmb

imgData = np.load("imgData.pickle", allow_pickle=True)

# create baker object
baker = nmb.NavmeshBaker()
# add geometry, for example a simple plane
# the first array contains vertex positions, the second array contains polygons of the geometry

for x in range(4095):
    for y in range(2047):
        if imgData[y][x]:
            baker.add_geometry([(x, 0.0, y), (x, 0.0, y + 1.0), (x + 1.0, 0.0, y + 1.0), (x + 1.0, 0.0, y)], [[0, 1, 2, 3]])

# bake navigation mesh
baker.bake()
# obtain polygonal description of the mesh
# vertices, polygons = baker.get_polygonization()

baker.save_to_binary("map_mesh.bin")
```

**Comment by Tugcga** [comment]
May be the down-sampling of the input image will help? Original image ```4096 x 2048``` leads to more than 8 million polygons. Try to convert it to something like ```1024 x 512``` (500k polygons). But with this approach small islands from the map can disappear.

**Comment by Tugcga** [comment]
And one more comment. Be careful with ```cell_size``` parameter for baking. During the baking process the module voxelize the input mesh. If ```cell_size``` is equal to ```0.2``` (for example), then each square polygon ```1 x 1``` will be voxelized as 25 voxels. It can be too much for the Python.
